src/kernel/src/process/module/recompiler.rs

```diff
@@ -647,8 +660,22 @@ impl<'input> Recompiler<'input> {
     /// Get register other than `keep`.
     fn temp_register64(keep: Register) -> Register {
         match keep {
+            Register::R8 => Register::R9,
+            Register::R9 => Register::R8,
+            Register::R10 => Register::R11,
+            Register::R11 => Register::R10,
+            Register::R12 => Register::R13,
+            Register::R13 => Register::R12,
+            Register::R14 => Register::R15,
+            Register::R15 => Register::R14,
+            Register::RDI => Register::RSI,
+            Register::RSI => Register::RDI,
+            Register::RBP => Register::RSP,
```

Comment:
We cannot use `RSP` as a general purpose register because it must be pointing to the stack all the time.